src/Symfony/Component/Workflow/TransitionBlocker.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+/**
+ * A reason why a transition cannot be performed for a subject.
+ */
+class TransitionBlocker
+{
+    const REASON_CODE_TRANSITION_NOT_DEFINED = 'com.symfony.www.workflow.transition_blocker.not_defined';
+    const REASON_CODE_TRANSITION_NOT_APPLICABLE = 'com.symfony.www.workflow.transition_blocker.not_applicable';
+    const REASON_CODE_TRANSITION_UNKNOWN = 'com.symfony.www.workflow.transition_blocker.unknown';
+
+    private $message;
+    private $code;
+    private $parameters;
+
+    /**
+     * Creates a new transition blocker.
+     *
+     * @param string      $message    The blocker message
+     * @param string|null $code       The error code of the blocker
+     * @param array       $parameters The parameters that may be useful to pass
+     *                                around with the blocker
+     */
+    public function __construct($message, $code = null, array $parameters = array())
+    {
+        $this->message = $message;
+        $this->parameters = $parameters;
+        $this->code = $code;
+    }
+
+    /**
+     * Create a blocker, that says the transition cannot be made because it is undefined
+     * in a workflow.
+     *
+     * @param string $transitionName
+     * @param string $workflowName
+     *
+     * @return static
+     */
+    public static function createNotDefined(string $transitionName, string $workflowName)
+    {
+        return new static(
```

Comment:
Well, what's better:

a) allocating local variables, to avoid using newlines in 1 instruction (runtime overhead)
b) using newlines (no drawbacks)

I'll do what you asked for, though.